### PR TITLE
Allow uploading of pk7 files

### LINKED ISSUE #541: Allow users to upload pk7s
So, a couple of things:

* We will add a new endpoint for pk7s (because the backend doesn't get the extension)
* We will remove support for pkxs
* We will change _rawpk6 to _rawFile
* We will add either generation or extension to the pokemon to distinguish between them
* pk6parse will get a new parsing mode

### PR DESCRIPTION
**Pull request by YaManicKill**
Fixes #541

### CODE REVIEW

**Comment by YaManicKill** [comment]
Alright. I shall do some more testing on this today, but I think the only thing this is missing before we can merge it in is pkparse 1.0.0 being published.

**Comment by not-an-aardvark** [comment]
pkparse is now published, so I restarted the builds.

File: api/services/PokemonHandler.js [hunk 1]
```diff
@@ -159,7 +159,7 @@ exports.createPokemonFromFile = async ({user, visibility, boxId, file, gen = 6})
   parsed.owner = user.name;
   parsed.visibility = visibility;
 
-  // The next three lines will be a no-op after pk6parse outputs `gen`, and `_rawFile` instead of `_rawPk6`.
+  // The next three lines will be a no-op after pkparse outputs `gen`, and `_rawFile` instead of `_rawPk6`.
   parsed.gen = parsed.gen || gen;
   parsed._rawFile = parsed._rawFile || parsed._rawPk6;
   delete parsed._rawPk6;
```
Comment (not-an-aardvark) [inline api/services/PokemonHandler.js]: These three lines and the comment can be removed.

File: client/pokemon/pokemon.ctrl.js [hunk 2]
```diff
@@ -33,7 +33,7 @@ module.exports = class Pokemon {
   }
   parseBoxViewProps () {
     this.parsedOt = replace3dsUnicodeChars(this.data.ot);
-    this.paddedTid = this.data.tid.toString().padStart(5, '0');
+    this.paddedTid = (this.data.idNo || this.data.tid).toString().padStart(5, '0');
```
Comment (YaManicKill) [inline client/pokemon/pokemon.ctrl.js]: Ah yes, good plan.

File: test/controller/pokemon.js [hunk 3]
```diff
@@ -387,18 +387,18 @@ describe('PokemonController', () => {
...
-      // monkeypatch pk6parse.SUPPORTED_GENS to include 7.
+      // monkeypatch pkparse.SUPPORTED_GENS to include 7.
       before(() => {
-        pk6parse.SUPPORTED_GENS = pk6parse.SUPPORTED_GENS || [6, 7];
+        pkparse.SUPPORTED_GENS = pkparse.SUPPORTED_GENS || [6, 7];
       });
       after(() => {
-        delete pk6parse.SUPPORTED_GENS;
+        delete pkparse.SUPPORTED_GENS;
       });
```
Comment (YaManicKill) [inline test/controller/pokemon.js]: Alrighty.

File: api/services/PokemonHandler.js (hunk 1 above)
Comment (YaManicKill) [inline api/services/PokemonHandler.js]: o7

**Comment by YaManicKill** [comment]
Alright. Comments done. I'm going to not squash this one, because each commit is useful in itself, I think.

**Pull request merged**